rotkehlchen/chain/ethereum/utils.py

```diff
@@ -41,29 +41,59 @@
 MULTICALL_CHUNKS = 20
 
 
-def token_normalized_value_decimals(token_amount: int, token_decimals: int | None) -> FVal:
-    if token_decimals is None:  # if somehow no info on decimals ends up here assume 18
-        token_decimals = 18
+@overload
+def _normalize_value(
+        amount: int | FVal,
+        decimals: int | None,
+        as_int: Literal[True],
+) -> int:
+    ...
 
-    return token_amount / (FVal(10) ** FVal(token_decimals))
 
+@overload
+def _normalize_value(
+        amount: int | FVal,
+        decimals: int | None,
+        as_int: Literal[False],
+) -> FVal:
+    ...
 
-def normalized_fval_value_decimals(amount: FVal, decimals: int) -> FVal:
-    return amount / (FVal(10) ** FVal(decimals))
 
+def _normalize_value(
+        amount: int | FVal,
+        decimals: int | None,
+        as_int: bool,
+) -> FVal | int:
+    if decimals is None:  # if somehow no info on decimals ends up here assume 18
+        decimals = 18
 
-def token_raw_value_decimals(token_amount: FVal, token_decimals: int | None) -> int:
-    if token_decimals is None:  # if somehow no info on decimals ends up here assume 18
-        token_decimals = 18
+    value = amount / (FVal(10) ** FVal(decimals))
+    if as_int:
+        return value.to_int(exact=False)
 
-    return (token_amount * (FVal(10) ** FVal(token_decimals))).to_int(exact=False)
+    return value
 
 
-def token_normalized_value(
-        token_amount: int,
-        token: EvmToken,
-) -> FVal:
-    return token_normalized_value_decimals(token_amount, token.decimals)
+def token_normalized_value_decimals(token_amount: int, token_decimals: int | None) -> FVal:
+    return _normalize_value(amount=token_amount, decimals=token_decimals, as_int=False)
+
+
+def normalized_fval_value_decimals(amount: FVal, decimals: int) -> FVal:
+    return _normalize_value(amount=amount, decimals=decimals, as_int=False)
+
+
+def asset_normalized_value(amount: int, asset: CryptoAsset | EvmToken) -> FVal:
```

Comment:
Again here perhaps since this is a function used multiple times many times deep in the logic why not make two functions? asset_normalized_value and token_normalized_value as it was before to avoid the if?